### ISSUE TITLE
Tests for sentences/utils.rs

### ISSUE THREAD

**Issue by hargoniX** [AeroRust/nmea#25]
As per #11 we wanna ensure max test coverage, as of now sentences/utils.rs is not at all tested but shared across all sentence parsers and thus a source for very common bugs.

Should be quite easy to do and might be an easy way for an individual to look into tests and make a small open source commit unless someone really wants this in the crate quickly.

**Comment by elpiel** [comment]
@gflow33 is picking up this issue to work on it.
If you have any questions or caveats you found, you can open a Draft PR or write here.

**Comment by elpiel** [comment]
@gflow33 have you had any progress on this?

**Comment by gflow33** [comment]
Yes I am still on this.

**Comment by trkohler** [comment]
hello! is it still relevant?

**Comment by elpiel** [comment]
Yes it is and we just hit a case where one of the parsers panics. 

There are many places in the code where we could improve the error handling and also test coverage.
